exonum-java-binding/common/src/main/java/com/exonum/binding/common/proofs/DbKeyCompressedFunnel.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.common.proofs;
+
+import com.exonum.binding.common.hash.Funnel;
+import com.exonum.binding.common.hash.PrimitiveSink;
+import com.exonum.binding.common.proofs.map.DbKey;
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * A funnel for a database key. Puts the LEB128 compressed binary representation of the given
+ * database key into the sink.
+ */
+public enum DbKeyCompressedFunnel implements Funnel<DbKey> {
+  INSTANCE;
+
+  @Override
+  public void funnel(DbKey from, PrimitiveSink into) {
+    int bitsLength = from.getNumSignificantBits();
+    writeUnsignedLeb128(into, bitsLength);
+
+    // Perform division, rounding the result up
+    int wholeBytesLength = getWholeBytesKeyLength(bitsLength);
+    byte[] key = from.getKeySlice();
+    into.putBytes(key, 0, wholeBytesLength);
+  }
+
+  public static Funnel<DbKey> dbKeyCompressedFunnel() {
+    return INSTANCE;
+  }
+
+  @VisibleForTesting
+  static int getWholeBytesKeyLength(int bitsLength) {
+    return (bitsLength + Byte.SIZE - 1) / Byte.SIZE;
+  }
+
+  private static void writeUnsignedLeb128(PrimitiveSink into, int value) {
+    // As we encode number of significant bits in a database key which is [0; 256], three bytes
```

Comment:
Why? 256 takes 9 bits: 100000000. We encode 7 bits of the original value in a byte, therefore, it cannot take more than _ceil(9/7) = two bytes_.